src/imap/scan_folders.rs

```diff
@@ -0,0 +1,75 @@
+use std::time::Instant;
+
+use crate::{config::Config, context::Context, dc_tools::time};
+use anyhow::Context as _;
+
+use crate::error::Result;
+use crate::imap::Imap;
+use async_std::prelude::*;
+
+use super::{get_folder_meaning, get_folder_meaning_by_name, FolderMeaning};
+
+impl Imap {
+    pub async fn scan_folders(&mut self, context: &Context) -> Result<()> {
+        use crate::config::Config::*;
```

Comment:
The reason for using this wildcard import in the first place was to prevent rustfmt from spreading the code over even more lines:
```diff
-                if context.get_config(ConfiguredSentboxFolder).await.is_none() {
+                if context
+                    .get_config(Config::ConfiguredSentboxFolder)
+                    .await
+                    .is_none()
+                {
                     context
-                        .set_config(ConfiguredSentboxFolder, Some(folder.name()))
+                        .set_config(Config::ConfiguredSentboxFolder, Some(folder.name()))
                         .await?;
                 }
             } else if folder_name_meaning == FolderMeaning::Spam
-                && context.get_config(ConfiguredSpamFolder).await.is_none()
+                && context
+                    .get_config(Config::ConfiguredSpamFolder)
+                    .await
+                    .is_none()
             {
```
(rustfmt does not change the other lines)

but that's not too important to me, so I can also remove the wildcard import.